preview/Win7Msix/Win7MSIXInstaller/ValidateRequest.cpp

```diff
@@ -0,0 +1,88 @@
+#include <windows.h>
+
+#include <shlobj_core.h>
+#include <CommCtrl.h>
+
+#include "FilePaths.hpp"
+#include "ValidateRequest.hpp"
+#include "GeneralUtil.hpp"
+#include <TraceLoggingProvider.h>
+#include <VersionHelpers.h>
+
+const PCWSTR ValidateRequest::HandlerName = L"ValidateRequest";
+
+HRESULT ValidateRequest::ExecuteForAddRequest()
+{
+    RETURN_IF_FAILED(ParseDependencyInfoFromPackage());
+    RETURN_IF_FAILED(MatchTDFAndVersionToOS());
+    return S_OK;
+}
+
+HRESULT ValidateRequest::ParseDependencyInfoFromPackage()
+{
+    PackageInfo* packageInfo = m_msixRequest->GetPackageInfo();
+
+    ComPtr<IMsixDocumentElement> domElement;
+    RETURN_IF_FAILED(packageInfo->GetManifestReader()->QueryInterface(UuidOfImpl<IMsixDocumentElement>::iid, reinterpret_cast<void**>(&domElement)));
+
+    ComPtr<IMsixElement> element;
+    RETURN_IF_FAILED(domElement->GetDocumentElement(&element));
+
+    // Obtain the TargetDeviceFamily tag info
+    ComPtr<IMsixElementEnumerator> dependencyEnum;
+    RETURN_IF_FAILED(element->GetElements(
+        L"/*[local-name()='Package']/*[local-name()='Dependencies']/*[local-name()='TargetDeviceFamily']",
+        &dependencyEnum));
+
+    BOOL hc = FALSE;
+    RETURN_IF_FAILED(dependencyEnum->GetHasCurrent(&hc));
+    if (!hc)
+    {
+        TraceLoggingWrite(g_MsixTraceLoggingProvider,
+            "No Target device family Found",
+            TraceLoggingLevel(WINEVENT_LEVEL_ERROR));
+        return E_NOT_SET;
+    }
+
+    ComPtr<IMsixElement> dependencyElement;
+    RETURN_IF_FAILED(dependencyEnum->GetCurrent(&dependencyElement));
+
+    Text<wchar_t> targetDeviceFamilyName;
+    RETURN_IF_FAILED(dependencyElement->GetAttributeValue(L"Name", &targetDeviceFamilyName));
+    m_targetDeviceFamilyName = targetDeviceFamilyName.Get();
+
+    Text<wchar_t> minVersion;
+    RETURN_IF_FAILED(dependencyElement->GetAttributeValue(L"MinVersion", &minVersion));
+    m_minVersion = minVersion.Get();
+
+    return S_OK;
+}
+
+HRESULT ValidateRequest::MatchTDFAndVersionToOS()
+{
+    //Check for win 10 < 1709 first

```

Comment:
>Check for win 10 < 1709 first [](start = 6, length = 29)

if you go the VerifyVersionInfo route as suggested in my other comment, then this point is moot, but the way you have it here the OS version is windows10, so it should support apps that target windows7 (6.1.7601.0) as well. I think you meant to reverse the ordering and check Windows 7 SP1 first, which is most restrictive, then check Win8, then win10

Reply:
Used VerifyVersionInfo.